src/check/parse/Parser.zig

```diff
@@ -484,3 +475,7 @@ pub fn finishParseAssign(self: *Parser, base_indent: u16) ?IR.NodeStore.BodyIdx
         return body;
     }
 }
+
+pub fn addProblem(self: *Parser, diagnostic: IR.Diagnostic) void {
```

Comment:
When possible I'd like to force everything to call `addMalformed` to add the appropriate malformed node to the tree. It shouldn't be possible to add a diagnostic/problem without doing so.

Reply:
I'd like to talk to you about malformed...